ccm_web/server/appHandler.ts

```diff
@@ -0,0 +1,103 @@
+import path from 'path'
+
+import express from 'express'
+import type { Express, Request, Response, Router } from 'express'
+import { IdToken, Provider } from 'ltijs'
+import Database from 'ltijs-sequelize'
+
+import { Config } from './config'
+
+interface EnvOptions {
+  isDev: boolean
+  staticPath: string
+}
+
+class AppHandler {
+  private readonly config: Config
+  private readonly apiRouter: Router
+  private readonly envOptions: EnvOptions
+
+  constructor (
+    config: Config,
+    envOptions: EnvOptions,
+    apiRouter: Router
+  ) {
+    this.config = config
+    this.envOptions = envOptions
+    this.apiRouter = apiRouter
+  }
+
+  // ltijs docs: https://cvmcosta.me/ltijs/#/
+  async setupLTI (): Promise<Express> {
+    const { db, lti } = this.config
+
+    const dbPlugin = new Database(
+      db.name,
+      db.user,
+      db.password,
+      { host: db.host, dialect: 'postgres', logging: false }
+    )
+
+    const provider = Provider.setup(
+      lti.encryptionKey, // Key used to sign cookies and tokens
+      { plugin: dbPlugin },
+      {
+        // Reserved routes
+        appRoute: '/',
+        loginRoute: '/login',
+        keysetRoute: '/keys',
+        // Set secure to true if the testing platform is in a different domain and https is being used
+        // Set sameSite to 'None' if the testing platform is in a different domain and https is being used
+        cookies: { secure: true, sameSite: 'None' }
+      }
+    )
+
+    // Redirect to the application root after a successful launch
+    provider.onConnect(async (token: IdToken, req: Request, res: Response) => {
+      console.log(token.userInfo)
+      return res.sendFile(path.join(this.envOptions.staticPath, 'index.html'))
```

Comment:
Is Line#58 needed? since I see you are doing the same thing in Line#93-95? Felt redundant having in both places.

Reply:
I'll check.